server/src/test/java/io/druid/curator/announcement/AnnouncerTest.java

```diff
@@ -231,4 +231,29 @@ public void testLeavesBehindTurdlingsThatAlreadyExisted() throws Exception
     Assert.assertEquals(initialStat.getMzxid(), curator.checkExists().forPath(parent).getMzxid());
   }
 
+  @Test
+  public void testLeavesBehindTurdlingsWhenToldTo() throws Exception
+  {
+    curator.start();
+    Announcer announcer = new Announcer(curator, exec);
+
+    final byte[] billy = "billy".getBytes();
+    final String testPath = "/somewhere/test2";
+    final String parent = ZKPaths.getPathAndNode(testPath).getPath();
+
+    announcer.start();
+    try {
+      Assert.assertNull(curator.checkExists().forPath(parent));
+
+      announcer.announce(testPath, billy, false);
```

Comment:
sure, I had just copied an older test so I will fix that one too. It looks like 'announce' does background ops but 'unannounce' does foreground ops- so we need to wait for 'announce' but not 'unannounce'.